python_modules/dagster/dagster/_core/definitions/assets.py

```diff
@@ -1334,7 +1335,14 @@ def get_io_manager_key_for_asset_key(self, key: AssetKey) -> str:
         )[0].io_manager_key
 
     def get_resource_requirements(self) -> Iterator[ResourceRequirement]:
-        yield from self.node_def.get_resource_requirements()  # type: ignore[attr-defined]
+        if self.is_executable:
+            yield from self.node_def.get_resource_requirements()  # type: ignore[attr-defined]
+        else:
+            for key in self.keys:
+                yield ExternalAssetIOManagerRequirement(
+                    key=self.get_io_manager_key_for_asset_key(key),
+                    asset_key=key.to_string(),
+                )
```

Comment:
This exactly copies how non-observable source assets provide requirements.

Reply:
Can you add a comment to that effect?

Reply:
Also it's not exactly right since you have added a new class?